lib/rpc.ts

```diff
@@ -420,6 +438,16 @@ export class Rpc extends EventEmitter implements IForwarderDest {
     return this._sendResponse(call.reqId, result);
   }
 
+  private _onReadyMessage(): void {
+    this._deferSending = false;
+    try {
+      processQueue(this._inactiveSendQueue, this._sendMessageCB!.bind(this));
```

Comment:
This code is a near-duplicate of a similar chunk elsewhere, and the discrepancy between them seems a bug rather than a feature. I.e. if sendMessage throws here, the queued call will not be rejected.

Also, there is no guarantee that when Ready message is received, _sendMessageCB is set. We should not assume it. We queue messages as long as either (_sendMessageCB
== null) or (deferSend). When one of them becomes false, we need to check the other. This can be done with a single helper:
```
   _checkSendReady() {
      if (_sendMessageCB && !deferSending) {
        try { processQueue } ...
     }
   }
   on MsgReady: { this._deferSending = false; _checkSendReady(); }
   in setSendMessage(sendMessage, deferSending=false): { this._sendMessageCB = sendMessage; this._deferSeding = deferSending; _checkSendReady(); }
```

Reply:
> Also, there is no guarantee that when Ready message is received, _sendMessageCB is set

I disagree, with IChannel interface we're sure that _sendMessageCB is set when _onReadyMessage is called.

> if sendMessage throws here, the queued call will not be rejected.

Ok, fixed.